src/aap_eda/core/models/setting.py

```diff
@@ -0,0 +1,47 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import yaml
+from django.db import models
+
+from aap_eda.conf.registry import settings_registry
+from aap_eda.core.utils.crypto.base import SecretValue
+from aap_eda.core.utils.crypto.fields import EncryptedTextField
+
+__all__ = ("Setting",)
+
+ENCRYPTED_STRING = "$encrypted$"
+
+
+class Setting(models.Model):
+    class Meta:
+        db_table = "core_setting"
+
+    key = models.CharField(max_length=255)
```

Comment:
Since we are going always to fetch a setting by his name, I think the table should have an index for that field.